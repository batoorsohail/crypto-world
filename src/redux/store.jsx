import { configureStore } from '@reduxjs/toolkit';
import cryptoCoinsReducer from './coin/cryptoSlice';

const store = configureStore({
  reducer: {
    cryptoCoins: cryptoCoinsReducer,
  },
});

export default store;
