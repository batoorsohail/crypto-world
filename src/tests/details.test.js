import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import Detail from '../pages/Details';
import store from '../redux/store';

describe('Detail', () => {
  test('renders correctly with valid coin', () => {
    const mockCoin = {
      id: 'bitcoin',
      name: 'Bitcoin',
      icon: 'https://example.com/bitcoin.png',
      price: 50000,
      priceChange1h: 0.5,
      priceChange1d: -1.2,
      totalSupply: 21000000,
      availableSupply: 19000000,
      marketCap: 1000000000,
      volume: 500000000,
    };
    const { container } = render(
      <Provider store={store}>
        <Detail />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
