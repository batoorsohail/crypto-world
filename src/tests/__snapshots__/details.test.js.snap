// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsScreen renders correctly with valid coin 1`] = `
<div>
  <p>
    Please go back to the home page and choose coin
  </p>
</div>
`;
